rskj-core/src/main/java/co/rsk/peg/union/UnionBridgeSupportImpl.java

```diff
@@ -166,26 +169,25 @@ public UnionResponseCode requestUnionRbtc(Transaction tx, co.rsk.core.Coin amoun
         return UnionResponseCode.SUCCESS;
     }
 
-    private boolean isCallerNoUnionBridgeContractAddress(Transaction tx) {
+    private boolean isCallerUnionBridgeContractAddress(Transaction tx) {
         RskAddress unionBridgeContractAddress = getUnionBridgeContractAddress();
-        boolean isCallerNoUnionBridgeContractAddress = !tx.getSender(signatureCache).equals(unionBridgeContractAddress);
-        if (isCallerNoUnionBridgeContractAddress) {
+        boolean isCallerUnionBridgeContractAddress = tx.getSender(signatureCache).equals(unionBridgeContractAddress);
+        if (!isCallerUnionBridgeContractAddress) {
             String baseMessage = String.format("Caller is not the Union Bridge Contract Address. Caller address: %s, Union Bridge Contract Address: %s", tx.getSender(), unionBridgeContractAddress);
-            logger.warn(LOG_PATTERN, "isCallerNoUnionBridgeContractAddress", baseMessage);
+            logger.warn(LOG_PATTERN, "isCallerUnionBridgeContractAddress", baseMessage);
         }
-        return isCallerNoUnionBridgeContractAddress;
+        return isCallerUnionBridgeContractAddress;
     }
 
-    private boolean isInvalidAmount(co.rsk.core.Coin amountRequested) {
-        boolean isAmountNullOrGreaterThanZero = amountRequested == null || amountRequested.compareTo(co.rsk.core.Coin.ZERO) < 1;
-        if (isAmountNullOrGreaterThanZero) {
+    private boolean isValidAmount(co.rsk.core.Coin amountRequested) {
+        boolean isAmountNullOrLessThanZero = amountRequested == null || amountRequested.compareTo(co.rsk.core.Coin.ZERO) < 1;
+        if (isAmountNullOrLessThanZero) {
             logger.warn(
-                "[{isInvalidAmount}] Amount requested cannot be negative or zero. Amount requested: {}", amountRequested);
-            return true;
+                "[isValidAmount] Amount requested cannot be negative or zero. Amount requested: {}", amountRequested);
```

Comment:
Message says the amount cannot be 0 but the validation allows it

Reply:
Variable name updated